lib/redlock/client.rb

```diff
@@ -272,7 +296,12 @@ def lock_instances(resource, ttl, options)
         { validity: validity, resource: resource, value: value }
       else
         @servers.each { |s| s.unlock(resource, value) }
-        false
+
+        if errors.size >= @quorum
+          raise LockAcquisitionError.new("Too many Redis errors prevented lock acquisition", errors)
+        else
```

Comment:
no need for `else` here, just `end` the `if` block and have `false` as a last statement as it was before